src/app/components/Search/TableSearchBar.tsx

```diff
@@ -0,0 +1,133 @@
+import { FC, useEffect, useState } from 'react'
+import TextField from '@mui/material/TextField'
+import SearchIcon from '@mui/icons-material/Search'
+import HighlightOffIcon from '@mui/icons-material/HighlightOff'
+import InputAdornment from '@mui/material/InputAdornment'
+import { COLORS } from '../../../styles/theme/colors'
+import IconButton from '@mui/material/IconButton'
+import { useScreenSize } from '../../hooks/useScreensize'
+import WarningIcon from '@mui/icons-material/WarningAmber'
+import { typingDelay } from '../../../styles/theme'
+import Typography from '@mui/material/Typography'
+import { useTranslation } from 'react-i18next'
+import Button from '@mui/material/Button'
+import { CardEmptyState } from '../CardEmptyState'
+import { inputBaseClasses } from '@mui/material/InputBase'
+
+export interface TableSearchBarProps {
+  placeholder: string
+  warning?: string
+  value: string
+  onChange: (value: string) => void
+}
+
+export const TableSearchBar: FC<TableSearchBarProps> = ({ value, onChange, placeholder, warning }) => {
+  const { isTablet } = useScreenSize()
+
+  const [isProblemFresh, setIsProblemFresh] = useState(false)
+
+  useEffect(() => {
+    if (warning) {
+      const timeout = setTimeout(() => {
+        setIsProblemFresh(false)
+      }, typingDelay)
+      return () => clearTimeout(timeout)
+    } else {
+      setIsProblemFresh(true)
+    }
+  }, [warning])
+
+  const startAdornment = (
+    <InputAdornment
+      position="start"
+      disablePointerEvents // Pass clicks through, so it focuses the input
+    >
+      <SearchIcon sx={{ color: COLORS.grayDark, ml: -3 }} />
+    </InputAdornment>
+  )
+
+  const onClearValue = () => onChange('')
+
+  const endAdornment = (
+    <InputAdornment position="end">
+      {value ? (
+        <IconButton color="inherit" sx={{ mt: -3, mb: -3, mr: -1 }} onClick={onClearValue}>
+          <HighlightOffIcon />
+        </IconButton>
+      ) : (
+        <span style={{ width: '38px' }} />
```

Comment:
Do we need empty span here? Can it not just be null?

Reply:
We want a placeholder, so that width of the input field stays the same, independently of the presence or lack of the `X` in the end.